### PR TITLE
Pop ups on Inventory Overview

### LINKED ISSUE #352: Polygon on Inventory Overview
**Describe the feature you'd like to see included in Tree Mapper App.**
- New UI for Tree Inventory to show Polygon instead of a list of coordinates
- clicking on a particular location should show the Image of that coordinate and some info

**What are the use cases for this feature?**
- Visualise the summary of the entire Registration

**Examples of this feature or functionality.**
Our own Webapp

image (3)

**Feature Checklist**

- [ ] I have searched for existing issues and to the best of my knowledge this is not a duplicate
- [ ] I have included at least one example image if it could be helpful (optional)

### PR DESCRIPTION
**Pull request by tejassonar**
Fixes #352 

Changes in this pull request:
- Created popups for Inventory Overview Screen

@ankitecd

### CODE REVIEW

File: app/components/Common/Markers/index.tsx [hunk 1]
```diff
@@ -48,35 +77,83 @@ interface PointAnnotationMarkerProps {
...
+      // this._map.moveTo(feature.geometry.coordinates, 500);
+    }
+  };
+
   if (onePolygon.geometry.type === 'Point') {
     markers.push(
       <MapboxGL.PointAnnotation
         key={`${i}-point`}
         id={`${i}-point`}
+        title="Test"
```
Comment (norbertschuler) [inline app/components/Common/Markers/index.tsx:140]: The test "Test" does not show up in the UI, does it?

**Review by norbertschuler — CHANGES_REQUESTED** [review]
- [x] For a sample tree I can view the details and then close the screen coming back to the overview, but for an edge of the polygon a modal opens showing my photo (all test photos are black on iOS ;-)) and I have not found out how to close the dialog again - so I was stuck!

Simulator Screen Shot - iPhone 11 - 2021-08-11 at 16 17 12Simulator Screen Shot - iPhone 11 - 2021-08-11 at 16 17 25

**Comment by tejassonar** [comment]
> * [ ]  For a sample tree I can view the details and then close the screen coming back to the overview, but for an edge of the polygon a modal opens showing my photo (all test photos are black on iOS ;-)) and I have not found out how to close the dialog again - so I was stuck!
> 
> Simulator Screen Shot - iPhone 11 - 2021-08-11 at 16 17 12Simulator Screen Shot - iPhone 11 - 2021-08-11 at 16 17 25

This issue exists only on ios devices. On an android device, the modal will be closed if you tap outside the modal. Maybe harsh will debug this issue as he's the one who has the setup for Ios.

File: app/components/Common/Markers/index.tsx (hunk 1 above)
Comment (tejassonar) [reply app/components/Common/Markers/index.tsx:140]: No. it doesn't

**Comment by norbertschuler** [comment]
> This issue exists only on ios devices. On an android device, the modal will be closed if you tap outside the modal. Maybe harsh will debug this issue as he's the one who has the setup for Ios.

I inspected the app and the modal seems to cover the whole screen, so there is no outside area to click, is there?

Simulator Screen Shot - iPhone 11 - 2021-08-11 at 17 10 27

**Comment by tejassonar** [comment]
> > This issue exists only on ios devices. On an android device, the modal will be closed if you tap outside the modal. Maybe harsh will debug this issue as he's the one who has the setup for Ios.
> 
> I inspected the app and the modal seems to cover the whole screen, so there is no outside area to click, is there?
> 
> Simulator Screen Shot - iPhone 11 - 2021-08-11 at 17 10 27

Yes, there is (in android device). You may try it on the android device on BrowserStack. We used the same logic that we had used in the ProfileModal.

**Comment by norbertschuler** [comment]
> Yes, there is (in android device). You may try it on the android device on BrowserStack. We used the same logic that we had used in the ProfileModal.

I did not know the profile modal can be closed any other way as clicking the close button. And it also does not work on iOS trying to close it by clicking outside the modal area ;-)

File: app/components/InventoryOverview/index.tsx [hunk 2]
```diff
@@ -649,6 +883,174 @@ const InventoryOverview = ({ navigation }: any) => {
...
+  const heightUnit = nonISUCountries.includes(countryCode)
+    ? i18next.t('label.select_species_feet')
+    : 'm';
+  const diameterUnit = nonISUCountries.includes(countryCode)
+    ? i18next.t('label.select_species_inches')
+    : 'cm';
+  return (
+    <Modal visible={coordinateModalShow} transparent>
+      <TouchableOpacity
+        style={styles.outsideModalContainer}
```
Comment (norbertschuler) [inline app/components/InventoryOverview/index.tsx:957]: If this does not work, I suggest encapsulating the `Image` with `TouchableOpacity` to close the modal.

**Review by norbertschuler — APPROVED** [review]
I made some modification for the modal dialog now closing for every click (except the one opening the sample tree screen) instead of only clicks outside the dialog.

**Review by ankitecd — APPROVED** [review]
Thanks @norbertschuler looks good now

**Pull request merged**